### PR TITLE
Fix #5609 remove LegacyCoinTypeExtensions 

### LINKED ISSUE #5609: Wallet: Remove `LegacyCoinTypeExtension` file

### PR DESCRIPTION
**Pull request by nuo-xu**
## Summary of Changes
Remove `LegacyCoinTypeExtensions` and start fetching coin type via WalletService

This pull request fixes #5609 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: BraveWallet/Crypto/Stores/UserAssetsStore.swift [hunk 1]
```diff
@@ -155,7 +161,7 @@ public class UserAssetsStore: ObservableObject {
         withTimeInterval: 0.25, repeats: false,
         block: { [weak self] _ in
           guard let self = self else { return }
-          self.rpcService.network { network in
+          self.rpcService.network(.eth) { network in
```
Comment (StephenHeaps) [inline BraveWallet/Crypto/Stores/UserAssetsStore.swift:164]: Should we fetch selected coin from Wallet Service here?

File: BraveWallet/Crypto/Stores/SwapTokenStore.swift [hunk 2]
```diff
@@ -744,8 +760,8 @@ extension SwapTokenStore: BraveWalletJsonRpcServiceObserver {
     else { return }
     selectedFromToken = nil
     selectedToToken = nil
-    prepare(with: accountInfo) { [weak self] in
-      self?.fetchPriceQuote(base: .perSellAsset)
+    prepare(with: accountInfo) { [self] in
```
Comment (StephenHeaps) [inline BraveWallet/Crypto/Stores/SwapTokenStore.swift:763]: weak self no longer needed here?

File: BraveWallet/Crypto/Stores/UserAssetsStore.swift (hunk 1 above)
Comment (nuo-xu) [reply BraveWallet/Crypto/Stores/UserAssetsStore.swift:164]: based on the logic seems like we only want to check the token info for mainnet. so i just used .eth coin type directly.

File: BraveWallet/Crypto/Stores/SwapTokenStore.swift (hunk 2 above)
Comment (nuo-xu) [reply BraveWallet/Crypto/Stores/SwapTokenStore.swift:763]: dont think there is a retain cycle

**Review by StephenHeaps — APPROVED** [review]

**Pull request merged**